csrc/optimization/alias_analysis.cpp

```diff
@@ -0,0 +1,81 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <unordered_map>
+#include <vector>
+
+#include <fusion.h>
+#include <ir/interface_nodes.h>
+#include <ir/internal_base_nodes.h>
+#include <ir/utils.h>
+#include <optimization/alias_analysis.h>
+
+namespace nvfuser::optimization {
+
+namespace {
+
+// Returns whether the input TensorView is contiguous on every non-broadcast
+// IterDomain.
+bool isContiguous(const TensorView& tv) {
+  NVF_ERROR(tv.nDims() == tv.getContiguity().size());
+  for (const auto i : c10::irange(tv.nDims())) {
+    if (!tv.axis(static_cast<int>(i))->isBroadcast() &&
```

Comment:
Shouldnt this inspect allocation domain instead of leaf domain?

Reply:
Good question. Wdyt, @jjsjann123 ? In most cases, the allocation domain is empty. Do we treat empty as the same as rfactor domain? (To me, this makes more sense than "the same as leaf domain" because the leaf domain describes the schedule, which is orthogonal to the memory layout).

Reply:
yes and yes.
We were mistakenly using root/rfactor/leaf domain when we should have been using alloc domain instead. Sorry about the inconsistency in our code base.

I'm also patching this in TensorDomain as well as fusion record at this time. Hopefully we'll have them cleaned up slowly....

Reply:
PTAL the new logic.